src/sentry/lang/native/sources.py

```diff
@@ -173,6 +179,35 @@ def get_internal_artifact_lookup_source_url(project):
     )
 
 
+def get_bundle_index_urls(
+    project: Project, release: str | None, dist: str | None
+) -> Tuple[str | None, str | None]:
+    debug_id_index, url_index = None, None
+
+    base_url = get_internal_artifact_lookup_source_url(project)
+
+    def make_download_url(bundle: ArtifactBundleFlatFileIndex):
+        timestamp = int(bundle.date_added.timestamp() * 1000)
+        # FIXME: Symbolicator needs a better way to extract the `file_id` from the download url
+        return f"{base_url}?_last_timestamp={timestamp}&download=bundle_index/{bundle.id}"
```

Comment:
We need to be very careful here, since the date_added must always be monotonically increasing with respect to real upload sequence but since uploads can happen on multiple workers that might have equal time, I would prefer to have some sort of atomic versioning of indexes.

If we end up having two updates with the same date, we don’t have a deterministic way of defining the winner.

Reply:
I thought this would be used as cache busting key, but that won’t work either. Symbolicator would want to have immutable IDs for that.

Reply:
Could you elaborate a bit more?

My idea was to send to symbolicator a triple `project_id, release, dist` and a unique id, so that it knows that an update has to be performed if:
* Triple is not found in memory
* Triple exists and the id is lower